cmp/cmpopts/equate.go

```diff
@@ -87,3 +88,38 @@ func areNaNsF64s(x, y float64) bool {
 func areNaNsF32s(x, y float32) bool {
 	return areNaNsF64s(float64(x), float64(y))
 }
+
+// EquateApproxTime returns a Comparer options that
+// determine two time.Time values to be equal if they
+// are within the given time interval of one another.
+// Note that if both times have a monotonic clock reading,
+// the monotonic time difference will be used.
+//
+// The zero time is treated specially: it is only considered
+// equal to another zero time value.
+//
+// It will panic if margin is negative.
+func EquateApproxTime(margin time.Duration) cmp.Option {
+	if margin < 0 {
+		panic("negative duration in EquateApproxTime")
+	}
+	return cmp.FilterValues(func(x, y time.Time) bool {
+		return !x.IsZero() && !y.IsZero()
+	}, cmp.Comparer(timeApproximator{margin}.compare))
+}
+
+type timeApproximator struct {
```

Comment:
Might as well use `type timeApproximator time.Duration`, or just make the compare function get durqtion and return a function.

Reply:
I did the latter originally, but changed to use a struct [as requested](https://github.com/google/go-cmp/pull/158#discussion_r316572317) by @dsnet to make stack traces clearer. I think it's marginally clearer to have the margin as a field than making the type a `time.Duration` directly, so I'm inclined to leave as is. Happy to change it if you feel strongly.

Reply:
We can keep it as is. It's keeps consistency with the code above it.